Build foundry from source

sveitser opened the pull request:
I think I found a commit of foundry that works to run `make devnet-up-espresso` and `pnpm semver-lock` and `forge fmt`.

sveitser committed:
WIP: build foundry from source

It's still compiling ☕

sveitser commented on flake.nix:
```diff
@@ -5,42 +5,38 @@
+            go = prev."go_1_${toString goVersion}";
+            # Overlaying nodejs here to ensure nodePackages use the desired
+            # version of nodejs.
+            nodejs = prev.nodejs-18_x;
```
I had to update nixpkgs to get a new enough rust to build foundry but that doesn't contain nodejs-16 anymore. Not sure if updating to 18 breaks anything but we'll find out.

jbearer replied:
Wait isn't rust not coming from nixpkgs but from oxalica/rust-overlay?

sveitser replied:
I haven't looked into how to make buildRustPackage work with the oxalica overlay. It might be possible but since we don't need a rust toolchain in the dev shell this was easier. Upstream uses node v20 anyway so we might as well switch to that too.

sveitser committed:
Potentially working commit of foundry

sveitser committed:
Disable geth version check

- Use nodejs 20 like nvmrc.

jbearer committed:
Reenable geth-version-checker for CI

sveitser committed:
dev shell: pin geth 1.13.4

This matches the current .gethrc version. Updating the geth version is a
bit inconvenient but instructions are included as comments.
